From 227afd2acd1786bfb3c56364513ea6727abc212f Mon Sep 17 00:00:00 2001
From: "cheng.chen" <c392410997@gmail.com>
Date: Fri, 11 Nov 2022 12:13:42 +0800
Subject: [PATCH] Adaption for Iluvatar

---
 torch/CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/torch/CMakeLists.txt b/torch/CMakeLists.txt
index 29851a9..42f941c 100644
--- a/torch/CMakeLists.txt
+++ b/torch/CMakeLists.txt
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.4)
 project(torch_cpp_impl)
 
-find_package(Torch 1.10 EXACT REQUIRED)
+find_package(Torch 1.10 REQUIRED)
 if (Torch_FOUND)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O2 -std=c++14 -fPIC")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${TORCH_CXX_FLAGS}")
@@ -11,6 +11,7 @@ if (Torch_FOUND)
     add_definitions(-DTORCH_VERSION_MINOR=${Torch_VERSION_MINOR})
     add_definitions(-DTORCH_VERSION_PATCH=${Torch_VERSION_PATCH})
     add_definitions(-DTORCH_VERSION=${Torch_VERSION})
+    add_definitions(-D_GLIBCXX_USE_CXX11_ABI=0)
     message(STATUS "Found Torch Version: ${Torch_VERSION}")
 else()
     message(FATAL_ERROR "Libtorch version 1.10 is required, no suitable version was found")
-- 
2.34.1

